classes/class-admin.php

```diff
@@ -728,19 +759,165 @@ public function wp_ajax_reset() {
 	private function erase_stream_records() {
 		global $wpdb;
 
-		$where = '';
+		// If this is a multisite and it's not networked activated,
+		// only delete the entries from the blog which made the request.
+		if ( self::WP_STREAM_MULTI_NOT_NETWORK === $this->plugin->get_site_type() ) {
 
-		if ( is_multisite() && ! $this->plugin->is_network_activated() ) {
-			$where .= $wpdb->prepare( ' AND `blog_id` = %d', get_current_blog_id() );
+			// First check the log size.
+			$stream_log_size = self::get_blog_record_table_size();
+
+			// If this is a large log and we need to delete only the entries
+			// pertaining to an individual site, we will need to do those in batches.
+			if ( $this->plugin->is_large_records_table( (int) $stream_log_size ) ) {
+				$this->schedule_erase_large_records( $stream_log_size );
+				return;
+			}
+
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE `stream`, `meta`
+					FROM {$wpdb->stream} AS `stream`
+					LEFT JOIN {$wpdb->streammeta} AS `meta`
+					ON `meta`.`record_id` = `stream`.`ID`
+					WHERE 1=1 AND `blog_id`=%d;",
```

Comment:
@tharsheblows The `WHERE 1=1` is typically added when a SQL query is dynamically generated. While it will be ignored by the database, there's no need to have it in code in this case, as all the "where" clauses are static:

```suggestion
					`blog_id`=%d;",
```

Reply:
Commit https://github.com/xwp/stream/pull/1543/commits/e931533237950d63a23b67adfcd7870fcd955ed6 

Thank you! :)